pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -338,9 +338,26 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
+			firstlinkdstPath, err := os.Readlink(logSymlink)
 			if err != nil {
-				glog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+				glog.Errorf("Failed to resolve the log symlink %q: %v", logSymlink, err)
+				return nil
+			}
+			secondlinkdstPath, err := os.Readlink(firstlinkdstPath)
```

Comment:
The second symlink is actually introduced in dockershim. While this is not alway true for all container runtimes, could we fix this inside dockershim package?

Reply:
By your suggestion, it's better to modify the "dockershim / removeContainerLogSymlink" function?https://github.com/kubernetes/kubernetes/blob/23226c24d4fd11bcb7aafda128aecbe3de3a0732/pkg/kubelet/dockershim/docker_container.go#L223

Reply:
@warmchang This function won't be called if container was removed by "docker rm".

Reply:
@feiskyer 
It may be a little hard to fix this inside dockershim package, as metioned above, "dockershim / removeContainerLogSymlink"  won't be called if container was removed by "docker rm".
Only the evictPodLogsDirectories function can make it up, what do you think if I add a judgement, my changes are applied only for docker,while for other container runtimes keep the current treatment？